### PR TITLE
Fix Call Graph Construction Bug

### PR DESCRIPTION
**Pull request by matthewbdwyer**
Removed creation of call graph edges for ASTVariables.  
They should only be created for ASTFunAppExpr nodes.  
Updated unit tests.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -77,9 +77,9 @@ cmake_policy(POP)
 ######## LLVM rules for installed libraries ###########
 # We want to select the latest stable release even if others are installed
 if(CMAKE_SYSTEM_NAME STREQUAL "Linux")
-  find_package(LLVM 14.0.0 REQUIRED CONFIG)
+  find_package(LLVM 14.0 REQUIRED CONFIG)
 elseif(CMAKE_SYSTEM_NAME STREQUAL "Darwin")
-  find_package(LLVM 14.0.0 REQUIRED CONFIG)
+  find_package(LLVM 14.0 REQUIRED CONFIG PATHS /usr/local/opt)
```
Comment (nicholasphair) [inline CMakeLists.txt:82]: Do we need to hardcode this path?

**Comment by nicholasphair** [comment]
Changes LGTM. It did not pass the ubuntu build though and I don't see any logs why. I'll try rerunning the job

File: CMakeLists.txt (hunk 1 above)
Comment (matthewbdwyer) [reply CMakeLists.txt:82]: No.  I think this should be setup in the shell path by bootstrap - which I think it is.   It was just a workaround for a situation where I didn't want to rerun bootstrap and I forgot to revert it before pushing.

Comment (nicholasphair) [reply CMakeLists.txt:82]: Okay, would you like me to remove it before merging or would that break things for you?

Comment (nicholasphair) [reply CMakeLists.txt:82]: Okay, would you like me to remove it before merging or would that break things for you?

**Review by matthewbdwyer — COMMENTED** [review]
LGTM

**Pull request merged**